From 8546c6afbdb821ebabdedf6ac74ae1ab28bf4927 Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@denx.de>
Date: Thu, 26 Mar 2020 15:14:11 +0100
Subject: [PATCH 30/44] ARM: dts: stm32: Add DT overlays for STM32MP15xx DHCOR
 Avenger96

Add DT overlays to support:
 - FDCAN1 on low-speed expansion X6
 - FDCAN2 on low-speed expansion X6
 - AT24C04 I2C EEPROM on low-speed expansion X6 I2C1
 - AT24C04 I2C EEPROM on low-speed expansion X6 I2C2
 - AT25AA010A SPI EEPROM on low-speed expansion X6 SPI2

Upstream-Status: Inappropriate
Signed-off-by: Marek Vasut <marex@denx.de>
---
 arch/arm/boot/dts/Makefile                    |  5 ++++
 ...tm32mp15xx-avenger96-overlay-fdcan1-x6.dts | 10 +++++++
 ...tm32mp15xx-avenger96-overlay-fdcan2-x6.dts | 10 +++++++
 ...p15xx-avenger96-overlay-i2c1-eeprom-x6.dts | 17 +++++++++++
 ...p15xx-avenger96-overlay-i2c2-eeprom-x6.dts | 17 +++++++++++
 ...p15xx-avenger96-overlay-spi2-eeprom-x6.dts | 28 +++++++++++++++++++
 6 files changed, 87 insertions(+)
 create mode 100644 arch/arm/boot/dts/stm32mp15xx-avenger96-overlay-fdcan1-x6.dts
 create mode 100644 arch/arm/boot/dts/stm32mp15xx-avenger96-overlay-fdcan2-x6.dts
 create mode 100644 arch/arm/boot/dts/stm32mp15xx-avenger96-overlay-i2c1-eeprom-x6.dts
 create mode 100644 arch/arm/boot/dts/stm32mp15xx-avenger96-overlay-i2c2-eeprom-x6.dts
 create mode 100644 arch/arm/boot/dts/stm32mp15xx-avenger96-overlay-spi2-eeprom-x6.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index e0f166a99b2e0..dac149ebd2dc8 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -1222,6 +1222,11 @@ dtb-$(CONFIG_ARCH_STM32) += \
 	stm32mp153c-dhcor-drc-compact.dtb \
 	stm32mp157a-avenger96.dtb \
 	stm32mp157a-dhcor-avenger96.dtb \
+	stm32mp15xx-avenger96-overlay-fdcan1-x6.dtbo \
+	stm32mp15xx-avenger96-overlay-fdcan2-x6.dtbo \
+	stm32mp15xx-avenger96-overlay-i2c1-eeprom-x6.dtbo \
+	stm32mp15xx-avenger96-overlay-i2c2-eeprom-x6.dtbo \
+	stm32mp15xx-avenger96-overlay-spi2-eeprom-x6.dtbo \
 	stm32mp157a-dk1.dtb \
 	stm32mp157a-dk1-scmi.dtb \
 	stm32mp157a-iot-box.dtb \
diff --git a/arch/arm/boot/dts/stm32mp15xx-avenger96-overlay-fdcan1-x6.dts b/arch/arm/boot/dts/stm32mp15xx-avenger96-overlay-fdcan1-x6.dts
new file mode 100644
index 0000000000000..d4f565d9e73c9
--- /dev/null
+++ b/arch/arm/boot/dts/stm32mp15xx-avenger96-overlay-fdcan1-x6.dts
@@ -0,0 +1,10 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) 2020 Marek Vasut <marex@denx.de>
+ */
+/dts-v1/;
+/plugin/;
+
+&m_can1 {
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/stm32mp15xx-avenger96-overlay-fdcan2-x6.dts b/arch/arm/boot/dts/stm32mp15xx-avenger96-overlay-fdcan2-x6.dts
new file mode 100644
index 0000000000000..12f51f75f264a
--- /dev/null
+++ b/arch/arm/boot/dts/stm32mp15xx-avenger96-overlay-fdcan2-x6.dts
@@ -0,0 +1,10 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) 2020 Marek Vasut <marex@denx.de>
+ */
+/dts-v1/;
+/plugin/;
+
+&m_can2 {
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/stm32mp15xx-avenger96-overlay-i2c1-eeprom-x6.dts b/arch/arm/boot/dts/stm32mp15xx-avenger96-overlay-i2c1-eeprom-x6.dts
new file mode 100644
index 0000000000000..bf9c3cbcdace3
--- /dev/null
+++ b/arch/arm/boot/dts/stm32mp15xx-avenger96-overlay-i2c1-eeprom-x6.dts
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) 2020 Marek Vasut <marex@denx.de>
+ */
+/dts-v1/;
+/plugin/;
+
+&i2c1 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	eeprom@56 {
+		compatible = "atmel,24c04";
+		reg = <0x56>;
+		pagesize = <16>;
+	};
+};
diff --git a/arch/arm/boot/dts/stm32mp15xx-avenger96-overlay-i2c2-eeprom-x6.dts b/arch/arm/boot/dts/stm32mp15xx-avenger96-overlay-i2c2-eeprom-x6.dts
new file mode 100644
index 0000000000000..a94b813255503
--- /dev/null
+++ b/arch/arm/boot/dts/stm32mp15xx-avenger96-overlay-i2c2-eeprom-x6.dts
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) 2020 Marek Vasut <marex@denx.de>
+ */
+/dts-v1/;
+/plugin/;
+
+&i2c2 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	eeprom@56 {
+		compatible = "atmel,24c04";
+		reg = <0x56>;
+		pagesize = <16>;
+	};
+};
diff --git a/arch/arm/boot/dts/stm32mp15xx-avenger96-overlay-spi2-eeprom-x6.dts b/arch/arm/boot/dts/stm32mp15xx-avenger96-overlay-spi2-eeprom-x6.dts
new file mode 100644
index 0000000000000..8f8395de36485
--- /dev/null
+++ b/arch/arm/boot/dts/stm32mp15xx-avenger96-overlay-spi2-eeprom-x6.dts
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) 2020 Marek Vasut <marex@denx.de>
+ */
+/dts-v1/;
+/plugin/;
+
+&spi2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&spi2_pins_a>;
+	status = "okay";
+	cs-gpios = <&gpioi 0 0>;
+	/* Use PIO for the 128 Byte SPI EEPROM */
+	dmas;
+	dma-names;
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	at25@0 {
+		compatible = "microchip,25aa010a", "atmel,at25";
+		reg = <0>;
+		spi-max-frequency = <5000000>;
+
+		at25,byte-len = <128>;
+		at25,addr-mode = <1>;
+		at25,page-size = <16>;
+	};
+};
-- 
2.42.0

